core/trino-grammar/src/main/antlr4/io/trino/grammar/sql/SqlBase.g4

```diff
@@ -720,11 +720,12 @@ booleanValue
     ;
 
 interval
-    : INTERVAL sign=(PLUS | MINUS)? string from=intervalField (TO to=intervalField)?
-    ;
-
-intervalField
-    : YEAR | MONTH | DAY | HOUR | MINUTE | SECOND
+    : INTERVAL sign=(PLUS | MINUS)? string from=YEAR (TO to=MONTH)?
+    | INTERVAL sign=(PLUS | MINUS)? string from=MONTH
+    | INTERVAL sign=(PLUS | MINUS)? string from=DAY (TO to=(HOUR | MINUTE | SECOND))?
+    | INTERVAL sign=(PLUS | MINUS)? string from=HOUR (TO to=(MINUTE | SECOND))?
+    | INTERVAL sign=(PLUS | MINUS)? string from=MINUTE (TO to=SECOND)?
+    | INTERVAL sign=(PLUS | MINUS)? string from=SECOND
```

Comment:
The rules basically say:

- If `TO` is specified then `from`SHALL NOT be `MONTH`
- If `TO` is specified then `from` SHALL be more significant than `to`.
- If `from=YEAR` then `to=MONTH` SHALL be the only valid option.

From the above the only valid combinations are:

```
from=YEAR
from=YEAR TO to=MONTH
from=MONTH
from=DAY
from=DAY TO to=(HOUR | MINUTE | SECOND)
from=HOUR
from=HOUR TO to=(MINUTE | SECOND)
from=MINUTE
from=MINUTE TO to=SECOND
from=SECOND
```

The changes seem to match this.